test/test_cli.rb

```diff
@@ -391,4 +391,14 @@ def test_silent
     assert_equal log_writer.stdout.class, Puma::NullIO
     assert_equal log_writer.stderr, $stderr
   end
+
+  def test_plugins
+    assert_empty Puma::Plugins.instance_variable_get(:@plugins)
+
+    cli = Puma::CLI.new ['--plugin', 'tmp_restart', '--plugin', 'systemd']
+    cli.send(:setup_options)
+
+    assert Puma::Plugins.instance_variable_get(:@plugins)["tmp_restart"]
+    assert Puma::Plugins.instance_variable_get(:@plugins)["systemd"]
```

Comment:
I think we can use the `#find` method over this?

https://github.com/puma/puma/blob/119330475ea62126d76b47fe202ac89dad1417dc/lib/puma/plugin.rb#L40